ml4gw/distributions.py

```diff
@@ -47,6 +48,59 @@ def __call__(self, N: int) -> torch.Tensor:
         return torch.arcsin(u / self.norm + math.sin(self.low))
 
 
+class CosineDistribution(dist.Distribution):
```

Comment:
I think for sake of naming consistency with the other `torch` distributions ( as well as our own ) we should name this `Cosine` instead of `CosineDistribution`

Reply:
Right, so that's something I wanted to ask. Note that the API is slightly different. For example, one has to call `.sample(...)` instead of the class object being callable. So downstream, one will have to do
```
cos = Cosine()
cos_samples = cos.sample((1000,))
```
instead of
```
cos = Cosine()
cos_samples = cos(10000)
```
But its a small enough change to be done in downstream libs.